api/src/main/java/edu/cornell/mannlib/vitro/webapp/controller/authenticate/ForgotPassword.java

```diff
@@ -0,0 +1,321 @@
+     */
+    private void notifyUser(UserAccount userAccount, I18nBundle i18n,
+                            VitroRequest vreq) {
+
+        Map<String, Object> body = new HashMap<String, Object>();
+        body.put("userAccount", userAccount);
+        body.put("passwordLink",
+            buildResetPasswordLink(userAccount.getEmailAddress(), userAccount.getEmailKey(), vreq));
+        body.put("siteName", vreq.getAppBean().getApplicationName());
+        body.put("subject", i18n.text("password_reset_pending_email_subject"));
+        body.put("textMessage", i18n.text("password_reset_pending_email_plain_text"));
+        body.put("htmlMessage", i18n.text("password_reset_pending_email_html_text"));
+
+        FreemarkerEmailMessage emailMessage = FreemarkerEmailFactory
+            .createNewMessage(vreq);
+        emailMessage.addRecipient(Message.RecipientType.TO, userAccount.getEmailAddress());
+        emailMessage.setBodyMap(body);
+        emailMessage.processTemplate();
+        emailMessage.send();
+    }
+
+    /**
+     * Generates a hash value for a given string.
+     *
+     * @param value The input string to be hashed.
+     * @return The computed hash value as a string.
+     */
+    private String captchaHash(String value) {
+        int hash = 5381;
+        value = value.toUpperCase();
+        for (int i = 0; i < value.length(); i++) {
+            hash = ((hash << 5) + hash) + value.charAt(i);
+        }
+        return String.valueOf(hash);
+    }
+
+    /**
+     * Sets common values in the data context for rendering templates.
+     *
+     * @param dataContext The data context to store common values.
+     * @param vreq        The VitroRequest object containing request information.
+     */
+    private void setCommonValues(Map<String, Object> dataContext, VitroRequest vreq) {
+        ApplicationBean appBean = vreq.getAppBean();
+
+        dataContext.put("forgotPasswordUrl", getForgotPasswordUrl(vreq));
+        dataContext.put("contactUrl", getContactUrl(vreq));
+        dataContext.put("emailConfigured", FreemarkerEmailFactory.isConfigured(vreq));
+        dataContext.put("contactEmailConfigured", StringUtils.isNotBlank(appBean.getContactMail()));
+        dataContext.put("wrongCaptcha", false);
+    }
+
+    /**
+     * Creates a response for indicating that an email has been successfully sent.
+     *
+     * @param dataContext The data context for rendering the response.
+     * @param i18n        The internationalization bundle for language translation.
+     * @param email       The email address to which the notification was sent.
+     * @return A ResponseValues object containing the email sent message.
+     */
+    private ResponseValues emailSentMessage(Map<String, Object> dataContext, I18nBundle i18n, String email) {
+        dataContext.put("message",
+            i18n.text("password_reset_email_sent") + email + i18n.text("password_reset_email_sent_if_exists"));
```

Comment:
I would advice not to concatenate translations as it not a good practice for translations. Instead provide email into translation string pattern. Example [usage](https://github.com/vivo-project/VIVO/blob/d23f3daab38a4a126b451d7748e01bfb3a796c84/webapp/src/main/webapp/templates/freemarker/body/createAndLinkResource/createAndLinkResourceEnterID.ftl#L17) and [translation](https://github.com/vivo-project/VIVO/blob/main/home/src/main/resources/rdf/i18n/sr_Latn_RS/interface-i18n/firsttime/vivo_UiLabel.ttl#L3785C19-L3785C19).